### PR TITLE
Increase chat timestamp resolution

### LINKED ISSUE #1545: Non-deterministic chat messages
When sending a series of closely timed chat messages to a peer, they can easily be presented out-of-order in the UI. We apply timestamps to the messages to ensure they can be deterministically ordered, however, the timestamps only have resolution to the second and it's not unreasonable for multiple chat messages to potentially arrive within the same second.

So we'd like to increase this resolution and understand what the limitation of the new resolution are to handle them appropriately in the future.

### PR DESCRIPTION
**Pull request by placer14**
Fixes #1545

### CODE REVIEW

File: repo/db/chat.go [hunk 1]
```diff
@@ -21,6 +21,12 @@ func (c *ChatDB) Put(messageId string, peerId string, subject string, message st
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// timestamp.UnixNano() is undefined when time has a zero value
+	if timestamp.IsZero() {
+		log.Errorf("putting chat message (%s): recieved zero timestamp, using current time", messageId)
```
Comment (cpacia) [inline repo/db/chat.go]: Maybe a warn here instead of an error?

File: repo/migrations/Migration023_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	if err = appSchema.BuildSchemaDirectories(); err != nil {
+		t.Fatal(err)
+	}
+	defer appSchema.DestroySchemaDirectories()
+
+	var (
+		databasePath = appSchema.DatabasePath()
+		schemaPath   = appSchema.DataPathJoin("repover")
+
+		//schemaSQL          = "pragma key = 'foobarbaz';"
```
Comment (cpacia) [inline repo/migrations/Migration023_test.go]: Maybe remove commented out var

File: repo/migrations/Migration023_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		CreateTableChatSQL = "create table chat (messageID text primary key not null, peerID text, subject text, message text, read integer, timestamp integer, outgoing integer);"
+		insertChatSQL      = "insert into chat(messageID, peerID, subject, message, read, timestamp, outgoing) values(?,?,?,?,?,?,?);"
+		selectChatSQL      = "select messageID, timestamp from chat;"
+		setupSQL           = strings.Join([]string{
```
Comment (cpacia) [inline repo/migrations/Migration023_test.go]: and update here

**Comment by cpacia** [comment]
Looks really good

**Comment by placer14** [comment]
We're going to move forward with this fix seeing that we have confirmed it properly resolves #1545. Need to update timestamps in the JSON API to ensure the full resolution can be shown in the string representation. Recommendation is to use RFC3339 which looks like `2006-01-02T15:04:05.999999999Z07:00`. Will create a new issue to track the changes needed. (FYI: @drwasho @jjeffryes)

**Pull request merged**